### PR TITLE
Switch username to walletname in new code

### PR DESCRIPTION
**Pull request by vindard**
## Description

We will be moving the naming reference from the user level to the wallet level. This PR is to switch all references of `username` to `walletname` in the new code, but to leave the old code intact for now so that now functionality changes or migrations are needed.

_**Note:** I didnt remove the `Username` type and `username` property our `User` type because that one goes all the way through to the new graphql schema._

### CODE REVIEW

File: src/domain/errors.ts
Comment (bodymindarts) [inline src/domain/errors.ts:14]: Is `InvalidUsername` still used somewhere?

File: src/domain/primitives/index.types.d.ts
Comment (bodymindarts) [inline src/domain/primitives/index.types.d.ts:17]: I'm wondering if we shouldn't use `WalletName` ie camel case

File: src/services/mongoose/wallets.ts
Comment (bodymindarts) [inline src/services/mongoose/wallets.ts:48]: from talking with @samerbuna about this a while ago I think using `null` for absence of a string is probably better.

**Review by bodymindarts — DISMISSED** [review]
In the name of moving forward I'm approving but have some non-critical suggestions. You should decide wether or not you want to address them / wait for another opinion or merge after another approval.

File: src/domain/primitives/index.types.d.ts
Comment (vindard) [reply src/domain/primitives/index.types.d.ts:17]: It's an easy change if we'd like to do this now

**Comment by dolcalmi** [comment]
did you consider how this change will impact mobile app? if we add a wallet name for each wallet then user/account will have to set it up (will be not only a graphic design problem but also a management problem for corporate accounts)

File: src/services/mongoose/wallets.ts
Comment (vindard) [reply src/services/mongoose/wallets.ts]: Done here: https://github.com/GaloyMoney/galoy/pull/475/commits/beec2a611a6434636fe0ae7d9ba5f59214b9bec7

File: src/domain/ledger/index.types.d.ts
Comment (dolcalmi) [inline src/domain/ledger/index.types.d.ts:32]: are you sure if you do this will not affect current data? I guess you will need to add a migration

**Review by dolcalmi — CHANGES_REQUESTED** [review]
@vindard please search for `.username` there are some pending changes

File: src/domain/ledger/index.types.d.ts
Comment (vindard) [reply src/domain/ledger/index.types.d.ts:32]: I was careful to isolation these `LedgerTransaction` changes to one commit (https://github.com/GaloyMoney/galoy/pull/475/commits/f5e7786575da2d77690b704273325b66a674ab36) to double-check that we don't need migrations.

So far, this type is only used in two places:
- input of `fromLedger` where it gets translated to `ConfirmedTransactionHistory` type so `username`-naming is irrelevant
- return from `getLiabilityTransactions`, which only gets used as the input to the `fromLedger` method

**Comment by vindard** [comment]
> @vindard please search for `.username` there are some pending changes

Those are all calls on `UserType` objects that come straight from our db. We haven't changed anything at the db level with this PR, only internal model-level changes. That `UserType` won't exist in the new code when we move all these instances across to using our own `User` types instead, accessed via the various repositories.

**Review by bodymindarts — APPROVED** [review]

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**
